src/main/java/org/passay/IllegalCharacterRule.java

```diff
@@ -11,7 +11,7 @@
  *
  * @author  Middleware Services
  */
-public class IllegalCharacterRule implements Rule
+public final class IllegalCharacterRule implements Rule
```

Comment:
Err, as mentioned above, this was also due to public non-final setter methods being used under the constructor.  I suppose I could also make the setters final instead.  Or have a private final both the constructor and the setters refer to.